server/app/controllers/partyController.js

```diff
@@ -9,99 +7,119 @@ class PartyController {
     if (!req.body.name) {
       res.status(400).send({
         status: 400,
-        error: 'Party name is required'
+        error: 'Party name is required',
       });
     } else if (!req.body.hqAddress) {
       res.status(400).send({
         status: 400,
-        error: 'HQ address is required'
+        error: 'HQ address is required',
       });
     } else if (!req.body.logoUrl) {
       res.status(400).send({
         status: 400,
-        error: 'Logo Url is required'
+        error: 'Logo Url is required',
       });
     } else {
       const party = {
         id: parties.length + 1,
         name: req.body.name,
         hqAddress: req.body.hqAddress,
-        logoUrl: req.body.logoUrl
+        logoUrl: req.body.logoUrl,
       };
       parties.push(party);
 
       res.status(201).send({
         status: 201,
-        party
+        data: [{
+          id: party.id,
+          name: party.name,
+        }],
       });
     }
   }
 
   getAllParties(req, res) {
+    const dataArray = [];
+    parties.forEach((party) => {
+      const data = {
+        id: party.id,
+        party: party.name,
+        logoUrl: party.logoUrl,
+      };
+      dataArray.push(data);
+    });
+
     res.status(200).send({
       status: 200,
-      parties
+      data: dataArray,
     });
   }
 
   getOneParty(req, res) {
-    const id = parseInt(req.params.id);
+    const id = parseInt(req.params.id, 10);
     parties.forEach((party) => {
       if (party.id === id) {
         res.status(200).send({
           status: 200,
-          party
+          data: [{
+            id: party.id,
+            name: party.name,
+            logoUrl: party.logoUrl,
+          }],
         });
       }
     });
     res.status(404).send({
       status: 404,
-      error: 'Resource not found'
+      error: 'Party does not exist',
     });
   }
 
   editParty(req, res) {
-    const id = parseInt(req.params.id);
-    const { name, hqAddress, logoUrl } = req.body;
+    const id = parseInt(req.params.id, 10);
+    const { name } = req.params;
```

Comment:
You are not getting the name from the params, you are getting it from the JSON body.

Reply:
Done